specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/linkedWorkspace.json

```diff
@@ -0,0 +1,88 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "AutomationManagement",
+    "version": "2015-10-31"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Automation/automationAccounts/{automationAccountName}/linkedWorkspace": {
+      "get": {
+        "tags": [
+          "LinkedWorkspace"
+        ],
+        "operationId": "LinkedWorkspace_Get",
+        "description": "Retrieve the linked workspace for the account id.",
+        "externalDocs": {
+          "url": "http://aka.ms/azureautomationsdk/linkedworkspaceoperations"
+        },
+        "x-ms-examples": {
+          "Get the linked workspace of an automation account": {
+            "$ref": "./examples/getLinkedWorkspace.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./definitions.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "automationAccountName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The automation account name."
+          },
+          {
+            "$ref": "./definitions.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./definitions.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./definitions.json#/definitions/LinkedWorkspace"
```

Comment:
This is a collection GET call. It should return a value array. Also it only has id property?

Reply:
Right now, our web service returns a single linked workspace, with id property only for GET linkedWorkspace. The id property will either be null or the resource id of the linked workspace.

Reply:
The API has been out there for 1.5 years. It is not on Azure REST API documentation because of an issue we had with available operations API, which has been fixed.

Reply:
well, what can we do then :). This is definitely not the correct design.

Reply:
Thanks @ravbhatnagar for your feedback. I will advocate fixing the API in our next version:

GET /providers/Microsoft.Automation/automationAccounts/{automationAccountName}/linkedWorkspace**s**

which returns an array of linked workspaces.

Of course nothing is set in stone yet.